engine/classes/ElggEntity.php

```diff
@@ -1670,6 +1697,8 @@ protected function update() {
 
 		_elgg_cache_entity($this);
 
+		$this->orig_attributes = [];
```

Comment:
Hmm, this needs to happen *after* the entity is completely stored, which may be after `ElggEntity::update`.